docs/appendices/release-notes/5.9.3.rst

```diff
@@ -109,3 +109,6 @@ Fixes
         ELSE 'default'
       END
     FROM test;
+
+- Fixed an issue that caused an ``UnsupportedFeatureException`` when validating
+  a complex ``ORDER BY`` expression.
```

Comment:
Since using position could lead to [the other bug](https://github.com/crate/crate/pull/16950#discussion_r1838308355), I think we should not promote it as a workaround?

Reply:
Right, let's only elaborate on the use case